### PR TITLE
Fix Gasteiger charges

### LINKED ISSUE #789: fitGasteigerCharges breaks on a molecule
[mymol.zip](https://github.com/Acellera/htmd/files/2456790/mymol.zip)

```python
from htmd.charge import fitGasteigerCharges
from htmd.molecule.molecule import Molecule
mol = Molecule('/tmp/mymol.mol2')
fitGasteigerCharges(mol)
```

```pytb
/shared/sdoerr/Work/htmd/htmd/charge/charge.py in fitGasteigerCharges(mol)
     63 
     64     # Compute and store Gasteiger charges
---> 65     ComputeGasteigerCharges(rdkit_mol, throwOnParamFailure=True)
     66     mol = mol.copy()
     67     mol.charge[:] = [atom.GetDoubleProp('_GasteigerCharge') for atom in rdkit_mol.GetAtoms()]

ValueError: ERROR: No Gasteiger Partial Charge parameters for Element: C Mode: 
```

### PR DESCRIPTION
**Pull request by raimis**
- [x] Implement `fitChargesWithAntechamber` (not used now, but it is left for AM1-BCC option)
- [x] Guess bond type if needed
  - [x] Implement `guessBontType`
- [x] Fix #789
- [x] Update tests

### CODE REVIEW

**Comment by stefdoerr** [comment]
Make it return any antechamber charges no? I'm using BCC currently

**Comment by raimis** [comment]
Yes, it can compute BCC! Although, BCC charges for water molecule look a bit suspicious in comparison with ESP and Gasteiger.

File: htmd/charge/charge.py [hunk 1]
```diff
@@ -69,6 +70,92 @@ def fitGasteigerCharges(mol):
     return mol
 
 
+def fitChargesWithAntechamber(mol, type='gas', molCharge=None):
```
Comment (stefdoerr) [inline htmd/charge/charge.py]: Are you using this somewhere? I could not find the reference.

File: htmd/charge/charge.py [hunk 2]
```diff
@@ -69,6 +70,92 @@ def fitGasteigerCharges(mol):
...
+        raise TypeError('"mol" has to be instance of {}'.format(Molecule))
+    if mol.numFrames != 1:
+        raise ValueError('"mol" can have just one frame, but it has {}'.format(mol.numFrames))
+
+    if type not in ('gas', 'bcc'):
+        raise ValueError('"type" has to be "gas" or "bcc"')
+
+    if molCharge is None:
+        molCharge = int(round(np.sum(mol.charge)))
+        logger.info('Using partial atomic charges to calculate molecular charge')
```
Comment (stefdoerr) [inline htmd/charge/charge.py]: You could also report the total charge here maybe? Could be useful for detecting issues. Although I think cli.py is already logging the charges?

File: htmd/charge/charge.py (hunk 1 above)
Comment (stefdoerr) [reply htmd/charge/charge.py]: Sorry, didn't see the comment above. Could you just add it as a BCC option for now? I need this to actually test my molecule dataset against parameterize

File: htmd/charge/charge.py (hunk 2 above)
Comment (raimis) [reply htmd/charge/charge.py]: Yes, CLI already reports

File: htmd/charge/charge.py (hunk 1 above)
Comment (raimis) [reply htmd/charge/charge.py]: AM1-BCC will be another PR. I also need implement symmetrization for them

**Pull request merged**